lib/index.js

```diff
@@ -15,16 +15,21 @@ WSTransport.register = function (transportProvider) {
 	transportProvider.registerTransport('wss', function () { return new WSTransport(); });
 };
 
-WSTransport.prototype.connect = function (address) {
+WSTransport.prototype.connect = function (address, sslOptions) {
 	// The subprotocol specified in extraHeaders is specific to Azure IoT Hub.
-	this._socket = ws.connect(address.href, { extraHeaders: { 'Sec-Websocket-Protocol': 'AMQPWSB10' } });
-	
+	if (!sslOptions) {
+	  this._socket = ws.connect(address.href, { extraHeaders: { 'Sec-Websocket-Protocol': 'AMQPWSB10' } });
+	} else {
+		sslOptions.extraHeaders = { 'Sec-Websocket-Protocol': 'AMQPWSB10' };
```

Comment:
if sslOptions already contains extraHeaders, they would be obliterated. Consider checking if extraHeaders are undefined and just add the additional property instead of clobbering the whole extraHeaders property